cli/src/commands/parallelize.rs

```diff
@@ -0,0 +1,163 @@
+    command: &CommandHelper,
+    args: &ParallelizeArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    // Parse and validate the input revset(s).
+    // Find the parents of each commit. The parent(s) which are ancestors of
+    // each revision are the new parents of the input revisions.
+    // Find the children of each commit. The children which are descendants
+    // of each revision are the new children of the input revisions.
+    // If one or both of these sets cannot be found, then parallelize cannot
+    // proceed.
+    // let target_commits = workspace_command.parse_union_revsets(&args.revisions)?.
+    let mut tx = workspace_command.start_transaction();
+    let target_revset = tx
+        .base_workspace_helper()
+        .parse_union_revsets(&args.revisions)?;
+    let target_commits: Vec<Commit> = tx
+        .base_workspace_helper()
+        .evaluate_revset(target_revset.clone())?
+        .iter()
+        // We want parents before children, so we need to reverse the order.
+        .reversed()
+        .commits(tx.base_repo().store())
+        .try_collect::<_, Vec<Commit>, _>()?;
+    if target_commits.len() < 2 {
+        return Ok(writeln!(ui.stderr(), "Nothing to do.")?);
+    }
+    let target_heads: Vec<CommitId> = tx
+        .base_workspace_helper()
+        .evaluate_revset(target_revset.clone().heads())?
+        .iter()
+        .collect();
+    if target_heads.len() > 1 {
+        let heads = target_heads.iter().map(short_commit_hash).join(", ");
+        return Err(user_error(format!(
+            "Cannot parallelize a set of revisions with multiple heads. Heads: {heads}"
+        )));
+    }
+    let target_roots: Vec<CommitId> = tx
+        .base_workspace_helper()
+        .evaluate_revset(target_revset.roots())?
+        .iter()
+        .collect();
+    if target_roots.len() > 1 {
+        let roots = target_roots.iter().map(short_commit_hash).join(", ");
+        return Err(user_error(format!(
+            "Cannot parallelize a set of revisions with multiple roots. Roots: {roots}"
+        )));
+    }
+
+    // Rebase the children of the head commit onto the target commits.
+    let new_children: Vec<Commit> = RevsetExpression::commit(target_heads[0].clone())
+        .children()
+        .evaluate_programmatic(tx.base_repo().as_ref())?
+        .iter()
+        .commits(tx.base_repo().store())
+        .try_collect()?;
+    for child in new_children {
+        // The ordering here is intentional. We iterate over the target commits first so
+        // that when `jj log` prints the graph they will be printed in their original
+        // order. After that, we add any existing parents which aren't being
+        // parallelized.
+        let get_id = |c: &Commit| c.id().clone();
+        let new_parents_for_child: Vec<Commit> = target_commits
+            .iter()
+            .map(get_id)
+            .chain(child.parents().iter().map(get_id))
```

Comment:
There is a `Commit::parent_ids()`

Reply:
😅 Thanks, I'll get the hang of it eventually.